test/saithrift/tests/copp_tests.py

```diff
@@ -0,0 +1,586 @@
+        self.routes.append(route_data)
+        return
+
+    def setup_route(self, ip_addr):
+
+        default_router_id = sai_thrift_get_default_router_id(self.client)
+        router_packet_action = SAI_PACKET_ACTION_FORWARD
+        cpu_port_next_hop = sai_thrift_get_cpu_port_id(self.client)
+
+        addr_family = SAI_IP_ADDR_FAMILY_IPV4
+        ip_addr1_subnet = ip_addr
+        ip_mask1 = '255.255.255.255'
+        sai_thrift_create_route(self.client, default_router_id, addr_family, ip_addr1_subnet, ip_mask1, cpu_port_next_hop, router_packet_action)
+        route_data=[default_router_id, addr_family, ip_addr1_subnet, ip_mask1, cpu_port_next_hop]
+        self.routes.append(route_data)
+        return
+
+    def create_routes(self):
+
+        self.setup_route(self.peerip)
+        if_suffix = int(self.test_params['test_port']) + 1
+        if_name = 'Ethernet%d' % if_suffix
+        sai_port_id = self.client.sai_thrift_get_port_id_by_front_port(if_name);
+        self.setup_direct_route(sai_port_id, self.peerip)
+        print'peerip==%s, adding direct route to if:%s'%(self.peerip,if_name)
+
+        return
+
+    @staticmethod
+    def cmd_run(cmdline):
+        cmd = cmdline.split(' ')
+        process = subprocess.Popen(cmd, stdout=subprocess.PIPE, stdin=subprocess.PIPE, stderr=subprocess.PIPE)
+        stdout, stderr = process.communicate()
+        ret_code = process.returncode
+
+        if ret_code != 0:
+            raise Exception("ret_code=%d, error message=%s. cmd=%s" % (ret_code, stderr, cmdline))
+        return stdout
+
+    def setUp(self):
+
+        ThriftInterfaceDataPlane.setUp(self)
+        self.policers=[]
+        self.trap_groups=[]
+        self.routes=[]
+        self.next_hops=[]
+        self.rifs=[]
+        return
+
+    def tearDown(self):
+        if config["log_dir"] != None:
+            self.dataplane.stop_pcap()
+
+        for route_data in self.routes:
+            sai_thrift_remove_route(self.client,route_data[0],route_data[1],route_data[2],route_data[3],route_data[4])
+        for nhop in self.next_hops:
+            self.client.sai_thrift_remove_next_hop(nhop)
+        for rif in self.rifs:
+            self.client.sai_thrift_remove_router_interface(rif)
+
+        for trap_gr in self.trap_groups:
+            self.client.sai_thrift_remove_hostif_trap_group(trap_group_id=trap_gr)
+
+        for trapid in self.trap_list:
+            sai_thrift_set_hostif_trap(self.client, trap_id=trapid, action=SAI_PACKET_ACTION_FORWARD)
+
+        for policer in self.policers:
+            status = self.client.sai_thrift_remove_policer(policer_id=policer)
+        return
+
+    def copp_test(self, packet, count, send_intf, recv_intf):
+
+        start_time=datetime.datetime.now()
+        for i in xrange(count):
+            testutils.send_packet(self, send_intf, packet)
+
+        end_time=datetime.datetime.now()
+
+        total_rcv_pkt_cnt = 0
```

Comment:
There are issues with count_matched_packets, this function can't resolve match_exp_pkt it calls internally.